Solutions/Corvus.Identity.ManagedServiceIdentity.ClientAuthentication/Corvus/Identity/ManagedServiceIdentity/ClientAuthentication/IServiceIdentityTokenSource.cs

```diff
@@ -49,7 +49,7 @@ public interface IServiceIdentityTokenSource
         /// service's MSI.
         /// </param>
         /// <returns>A task that produces an access token.</returns>
-        Task<string> GetAccessToken(string resource);
+        Task<string?> GetAccessToken(string resource);
```

Comment:
Is it really expected that this sometimes produces null? Is this being forced on us by `AzureServiceTokenProvider.GetAccessTokenAsync`?

It feels like an inability to get an access token would be an error. Or does the code that uses this actively check for null and handle it specially? (I know I wrote it originally, I just can't remember...)

Reply:
The latter - there is code explicitly checking if the result of this is null, so I assumed that it could be.